cpp/src/parquet/encoding-test.cc

```diff
@@ -314,6 +318,185 @@ TEST(TestDictionaryEncoding, CannotDictDecodeBoolean) {
   ASSERT_THROW(MakeDictDecoder<BooleanType>(nullptr), ParquetException);
 }
 
+// ----------------------------------------------------------------------
+// Shared arrow builder decode tests
+class TestDecodeArrow : public ::testing::Test {
+ public:
+  void SetUp() override {
+    InitDataInputs();
+    SetupEncoderDecoder();
+  }
+
+  void InitDataInputs() {
+    // Build a dictionary array and its dense representation for testing building both
+    auto dict_values = ::arrow::ArrayFromJSON(::arrow::binary(), "[\"foo\", \"bar\"]");
+    auto dtype = ::arrow::dictionary(::arrow::int8(), dict_values);
+    auto indices = ::arrow::ArrayFromJSON(::arrow::int8(), "[0, 1, 0, 0]");
+    ASSERT_OK(::arrow::DictionaryArray::FromArrays(dtype, indices, &expected_dict_));
+    expected_array_ =
+        ArrayFromJSON(::arrow::binary(), "[\"foo\", \"bar\", \"foo\", \"foo\"]");
+    num_values_ = static_cast<int>(expected_array_->length());
+
+    // arrow::BinaryType maps to parquet::ByteArray
```

Comment:
This is very minor, and I may not be understanding this correctly, but should this comment say `arrow::BinaryArray` instead of `arrow::BinaryType`?

Reply:
I can see that this is terse - the `expected_array_` is created with `arrow::DataType` equal to the `arrow::BinaryType`.  The loop is converting each value in the `expected_array_` into a `parquet::ByteArray`.

Reply:
Thanks for clarifying! That makes sense.